plugins/Files/js/reducers/files.js

```diff
@@ -0,0 +1,71 @@
+import { Map, List } from 'immutable'
+import * as constants from '../constants/files.js'
+import { ls, searchFiles } from '../sagas/helpers.js'
+
+const initialState = Map({
+	activespending: '',
+	allocatedspending: '',
+	usage: 0,
+	files: List(),
+	workingDirectoryFiles: List(),
+	searchResults: List(),
+	uploading: List(),
+	downloading: List(),
+	path: '',
+	searchText: '',
+	uploadSource: '',
+	showAllowanceDialog: false,
+	showUploadDialog: false,
+	showSearchField: false,
+	showFileTransfers: false,
+	showDeleteDialog: false,
+	dragging: false,
+})
+
+export default function filesReducer(state = initialState, action) {
+	console.log(action)
+	switch (action.type) {
+	case constants.RECEIVE_DISK_USAGE:
+		return state.set('usage', action.usage)
+	case constants.RECEIVE_FILES:
+		return state.set('files', action.files)
+                .set('workingDirectoryFiles', ls(action.files, state.get('path')))
+	case constants.SET_SEARCH_TEXT:
+		return state.set('searchResults', searchFiles(state.get('files'), action.text, state.get('path')))
+                .set('searchText', action.text)
```

Comment:
agreed, sublime shows the .sets as aligned but they need to have all spaces to be aligned properly iirc